### PR TITLE
Add highlights and source to inner hits

### PR DESCRIPTION
**Pull request by dbulaja98**

### CODE REVIEW

File: modules/library/src/main/scala/zio/elasticsearch/query/InnerHits.scala [hunk 1]
```diff
@@ -17,29 +17,113 @@
...
 
 import zio.Chunk
+import zio.elasticsearch.ElasticPrimitive.ElasticPrimitiveOps
+import zio.elasticsearch.highlights.Highlights
 import zio.json.ast.Json
-import zio.json.ast.Json.{Num, Obj, Str}
+import zio.json.ast.Json.{Arr, Num, Obj, Str}
 
 final case class InnerHits private[elasticsearch] (
+  private val excluded: Option[Chunk[String]],
```
Comment (markaya) [inline modules/library/src/main/scala/zio/elasticsearch/query/InnerHits.scala:26]: Why is this `[Option[Chunk]]`?

File: modules/library/src/test/scala/zio/elasticsearch/ElasticQuerySpec.scala [hunk 2]
```diff
@@ -2027,7 +2048,15 @@ object ElasticQuerySpec extends ZIOSpecDefault {
           val queryWithNested         = nested(TestDocument.subDocumentList, nested("items", term("testField", "test")))
           val queryWithIgnoreUnmapped = nested(TestDocument.subDocumentList, matchAll).ignoreUnmappedTrue
           val queryWithInnerHits =
-            nested(TestDocument.subDocumentList, matchAll).innerHits(InnerHits().from(0).size(3).name("innerHitName"))
+            nested(TestDocument.subDocumentList, matchAll).innerHits(
+              InnerHits()
+                .from(0)
+                .size(3)
+                .name("innerHitName")
+                .highlights(highlight("stringField"))
```
Comment (markaya) [inline modules/library/src/test/scala/zio/elasticsearch/ElasticQuerySpec.scala:2056]: Maybe we should consider creating highlight that accepts `TestDocument.stringField` as parameter

File: modules/library/src/test/scala/zio/elasticsearch/InnerHitsSpec.scala [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+import zio.json.ast.Json.Obj
+import zio.test.Assertion.equalTo
+import zio.test.{Spec, TestEnvironment, ZIOSpecDefault, assert}
+
+object InnerHitsSpec extends ZIOSpecDefault {
+
+  def spec: Spec[TestEnvironment, Any] =
+    suite("InnerHits")(
+      test("constructing") {
+        val innerHits               = InnerHits.apply()
```
Comment (markaya) [inline modules/library/src/test/scala/zio/elasticsearch/InnerHitsSpec.scala:17]: It should be without `apply`, just `InnerHits()`

**Review by markaya — CHANGES_REQUESTED** [review]

File: modules/library/src/test/scala/zio/elasticsearch/ElasticQuerySpec.scala (hunk 2 above)
Comment (dbulaja98) [reply modules/library/src/test/scala/zio/elasticsearch/ElasticQuerySpec.scala:2056]: I would do that to a separate PR, if that is ok with you.

**Pull request merged**